### ISSUE TITLE
Mongoose 6 support

### ISSUE THREAD

**Issue by demilsson** [florianholzapfel/express-restify-mongoose#432]
Would it be rather straight forward to add Mongoose 6 support? 
We are currently held back from being able to upgrade our database server to Mongo 5.

Many thanks for a great library!

**Comment by demilsson** [comment]
@Zertz I'm looking through the tests and most of the new failing tests introduced when using Mongoose 6 is related to error string formatting and in particular that `MongoError` is now called `MongoServerError` instead.

However, there was one test which I couldn't immediately dismiss where a populate command should be ignored and return a 200 status code. In Mongoose 6, trying to populate an undefined path now raises an error [unless populate is configured with the strictpopulate option set to false](https://mongoosejs.com/docs/migrating_to_6.html#strictpopulate).

So the question is whether the preference is to maintain the original behaviour of this library or to follow the new default behaviour of Mongoose... I would guess the former but it feels like it's a decision for a contributor with a better understanding of the code base to make.

The failing test:

```
  2) Express
       Read documents
         populate
           GET /Invoice?populate=evilCustomer 200 - ignore unknown field:

      Uncaught AssertionError [ERR_ASSERTION]: 400 == 200
      + expected - actual

      -400
      +200
      
      at Request._callback (test/integration/read.js:843:20)
      at Request.self.callback (node_modules/request/request.js:185:22)
      at Request.emit (node:events:520:28)
      at Request.<anonymous> (node_modules/request/request.js:1154:10)
      at Request.emit (node:events:520:28)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:1076:12)
      at Object.onceWrapper (node:events:639:28)
      at IncomingMessage.emit (node:events:532:35)
      at endReadableNT (node:internal/streams/readable:1346:12)
      at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

The actual response:

``` js
{
  message: 'Cannot populate path `evilCustomer` because it is not in your schema. Set the `strictPopulate` option to false to override.',
  name: 'MongooseError'
}
```

**Comment by Zertz** [comment]
First off, I'm glad the tests caught that!

> So the question is whether the preference is to maintain the original behaviour of this library or to follow the new default behaviour of Mongoose... I would guess the former

I'm honestly a bit torn here. I like what Mongoose is doing and it definitely makes sense as it makes it obvious you're trying to populate a field you shouldn't be. However, since we're exposing stuff to arbitrary clients, we should probably avoid implicitly providing too much information about the schema.

So yes, I agree that we should stick to the current behavior. If it's ever an issue, we can revisit.

**Comment by demilsson** [comment]
Great, thanks for taking a look! I'll work on preparing a PR.

**Comment by trannel** [comment]
Are there any updates on this, by any chance?

**Comment by demilsson** [comment]
Unfortunately not. I made a PR but there was a few existing test failures that I didn't want to touch, perhaps that's what holding it up?

**Comment by Zertz** [comment]
I merged it and created a followup PR to fix tests and clean up a few things. We should be able to publish a release sometimes soon.

**Comment by Zertz** [comment]
Oh, I thought it was released. I'll re-open this issue has a reminder and try to get it done this weekend.

**Comment by Zertz** [comment]
Published as 7.0.2
